src/Shared/NativeMethodsShared.cs

```diff
@@ -632,6 +632,7 @@ internal static bool IsMaxPathLegacyWindows()
             }
         }
 
+#pragma warning disable CA1416
```

Comment:
Instead of doing this, can you follow the guidance on this warning? https://docs.microsoft.com/en-us/dotnet/core/compatibility/code-analysis/5.0/ca1416-platform-compatibility-analyzer#recommended-action

Specifically, I'd expect to see this call annotated with `[SupportedOSPlatform("windows")]`, which will require annotating or conditionalizing up the chain.

Reply:
The guidance I followed is right below the guidance you suggested.

Also, [SupportedOSPlatform("windows")] was introduced in .net 6, as was OperatingSystem.IsWindows(), so I would have to ifdef around the attribute or IsWindows check. That's pretty terrible, so I don't think that's a good idea.

Reply:
At the very least this needs a comment.

Reply:
I can add one. It doesn't change anything, but I should correct myself: it was introduced in .net 5, not .net 6.